Changes

```diff
@@ -15,6 +15,14 @@ Working version
   use String_val as a char* instead of const char*
   (Kate Deplaix)
 
+- #8993: New C function caml_process_pending_events* in
+  caml/signals.h, intended for executing asynchronous callbacks
+  (signal handlers, finalisers, and memprof callbacks) in long-running
+  C functions. The function caml_process_pending_events_exn returns
+  any exception arising during their execution, so that resources can
+  be cleaned-up before re-raising it.
+  (Guillaume Munch-Maccagnoni, review by TODO)
+
```

Comment:
You also need to fix the doc you wrote for #8941, since `caml_check_urgent_gc` is now deprecated.

Reply:
Done